nucypher/blockchain/eth/actors.py

```diff
@@ -75,6 +83,146 @@ def token_balance(self):
         return balance
 
 
+class Deployer(NucypherTokenActor):
+
+    __interface_class = BlockchainDeployerInterface
+
+    def __init__(self,
+                 blockchain,
+                 deployer_address: str = None,
+                 token_agent: NucypherTokenAgent = None,
+                 miner_agent: MinerAgent = None,
+                 policy_agent: PolicyAgent = None,
+                 ) -> None:
+
+        self.__deployer_address = deployer_address or NO_DEPLOYER_ADDRESS
+        self.blockchain = blockchain
+
+        self.token_agent = token_agent or NO_CONTRACT_AVAILABLE
+        self.miner_agent = miner_agent or NO_CONTRACT_AVAILABLE
+        self.policy_agent = policy_agent or NO_CONTRACT_AVAILABLE
+
+        self.user_escrow_deployers = dict()
+
+    @classmethod
+    def from_deployed_blockchain(cls, provider_uri, *args, **kwargs) -> 'Deployer':
+        blockchain = cls.connect_to_blockchain(provider_uri=provider_uri)
+        token_agent = NucypherTokenAgent(blockchain=blockchain)
+        miner_agent = MinerAgent(blockchain=blockchain, token_agent=token_agent)
+        policy_agent = PolicyAgent(blockchain=blockchain, miner_agent=miner_agent)
+        instance = cls(blockchain=blockchain,
+                       token_agent=token_agent,
+                       miner_agent=miner_agent,
+                       policy_agent=policy_agent,
+                       *args, **kwargs)
+        return instance
+
+    @classmethod
+    def from_blockchain(cls, provider_uri: str, registry=None, *args, **kwargs):
+        blockchain = cls.connect_to_blockchain(provider_uri=provider_uri, registry=registry)
+        instance = cls(blockchain=blockchain, *args, **kwargs)
+        return instance
+
+    @classmethod
+    def connect_to_blockchain(cls, provider_uri: str, compile=True, registry=None):
+        compiler = SolidityCompiler() if compile else None
```

Comment:
Interesting pattern.  I imagine it occurred to you to make this a `classmethod` on `Blockchain`?  But this is good stuff I think.